File: homestay-backend/Homestay.Api/Domain/Entities/Room.cs
using System;
using System.Collections.Generic;

namespace Homestay.Api.Domain.Entities;

public partial class Room
{
    public Guid Id { get; private set; }

    public Guid HomestayId { get; set; }

    public string RoomName { get; set; } = null!;

    public decimal BasePrice { get; set; }

    public int Capacity { get; set; }

    public string Status { get; set; } = null!;

    public DateTime? CreatedAt { get; private set; }

    public virtual ICollection<Booking> Bookings { get;  set; } = new List<Booking>();

    public virtual Homestay Homestay { get;  set; } = null!;

    public virtual ICollection<RoomAvailability> RoomAvailabilities { get;  set; } = new List<RoomAvailability>();

    public virtual ICollection<Amenity> Amenities { get;  set; } = new List<Amenity>();

    public virtual ICollection<Promotion> Promotions { get;  set; } = new List<Promotion>();
}

File: homestay-frontend/types/room.ts
import { Amenity } from "./amenity";

export type RoomStatus = "AVAILABLE" | "UNAVAILABLE";

export interface Room {
  id: string;
  roomName: string;
  homestayId: string;
  homestayName: string;
  homestayAddress: string;
  homestayCity: string;
  capacity: number;
  basePrice: number;
  status: RoomStatus;
  createdAt: string;
  amenities: Amenity[];
}

export interface CreateRoomRequest {
  homestayId: string;
  roomName: string;
  capacity: number;
  basePrice: number;
  status: RoomStatus;
  amenityIds?: number[];
}

export interface UpdateRoomRequest {
  roomName?: string;
  capacity?: number;
  basePrice?: number;
  status?: RoomStatus;
  amenityIds?: number[];
}

export interface RoomResponse {
  id: string;
  roomName: string;
  homestayId: string;
  homestayName: string;
  homestayAddress: string;
  homestayCity: string;
  capacity: number;
  basePrice: number;
  status: RoomStatus;
  createdAt: string;
  amenities: Amenity[];
}
